supervisor/iptablesctrl/uidpam/update_portsets.go

```diff
@@ -0,0 +1,260 @@
+package uipam
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/user"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aporeto-inc/trireme-lib/cache"
+	"github.com/bvandewalle/go-ipset/ipset"
+	"go.uber.org/zap"
+)
+
+const (
+	procNetTCPFile                    = "/proc/net/tcp"
+	portSetUpdateIntervalMilliseconds = 1000
+	portEntryTimeout                  = 60
+	uidFieldOffset                    = 7
+	procHeaderLineNum                 = 1
+	portOffset                        = 1
+	ipPortOffset                      = 1
+	sockStateOffset                   = 3
+	sockListeningState                = "0A"
+	hexFormat                         = 16
+	integerSize                       = 64
+	decFormat                         = 10
+)
+
+// PortSet : This provides an interface to update the
+// look up table required to program the ipset portsets.
+type PortSet interface {
+	AddToUIDToPortSetCache(uid string, value string) (err error)
+	GetFromUIDToPortSetCache(uid string) (err error)
+	AddToUIDToPortsCache(uid string, value string) (ok bool)
+	RemoveFromUIDToPortSetCache(uid string) (err error)
+	RemoveFromUIDToPortsCache(uid string) (err error)
+	UpdatePortSet(uid string, port string)
+}
+
+// UIDCacheInstance : This type contains look up tables
+// to help update the ipset portsets.
+type UIDCacheInstance struct {
+	UIDToPortSet *cache.Cache
+	UIDToPorts   *cache.Cache
+	UIDSet       map[string]bool
+}
+
+// NewInstance : creates a new UIDCache instance
```

Comment:
Where is the IPSet programming Interface ? Not passed in ? How do we program the ipsets ?

Reply:
UpdatePortSet is the interface. I will change the name to AddPortset